### PR TITLE
feat: TextField editable in date-picker

### PR DESCRIPTION
**Pull request by singh-pk**
### Changelog [STD-1873]

**TODO:** 
- Make date-picker field as editable and allow user to either type string dates or in the formats DD-MM-YYYY / DD/MM/YYYY
- Convert date back to current format once user ends date selection

**Screenshot:**

https://github.com/user-attachments/assets/b8c708ab-f912-4574-bec6-eb400dd81da2

### CODE REVIEW

File: packages/raystack/calendar/date-picker.tsx [hunk 1]
```diff
@@ -2,71 +2,162 @@ import { Popover } from "~/popover";
...
   calendarProps?: PropsSingleRequired & PropsBase;
   onSelect?: (date: Date) => void;
+  placeholder?: string;
   value?: Date;
 }
 
 export function DatePicker({
   side = "top",
   dateFormat = "DD/MM/YYYY",
+  placeholder = 'DD/MM/YYYY',
```
Comment (paanSinghCoder) [inline packages/raystack/calendar/date-picker.tsx:29]:
```suggestion
  placeholder = "DD/MM/YYYY",
```

File: packages/raystack/calendar/date-picker.tsx [hunk 2]
```diff
@@ -2,71 +2,162 @@ import { Popover } from "~/popover";
...
           {...textFieldProps}
-        />
-      </Popover.Trigger>
-      <Popover.Content side={side} className={styles.calendarPopover}>
+      />
+
+      <Popover.Content side={side} className={styles.calendarPopover} ref={contentRef}>
         <Calendar
           required={true}
           {...calendarProps}
```
Comment (paanSinghCoder) [inline packages/raystack/calendar/date-picker.tsx:154]: {...calendarProps} - We can make this the last prop.

File: packages/raystack/calendar/date-picker.tsx [hunk 3]
```diff
@@ -2,71 +2,162 @@ import { Popover } from "~/popover";
...
+
+  function handleKeyUp(event: React.KeyboardEvent) {
+    if (event.code === 'Enter' && textFieldRef.current) {
+        textFieldRef.current.blur();
+        removeEventListeners();
+    }
+  }
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const { value } = (event.target);
```
Comment (paanSinghCoder) [inline packages/raystack/calendar/date-picker.tsx:116]: (event.target) - Why in bracket?

File: packages/raystack/calendar/date-picker.tsx (hunk 2 above)
Comment (singh-pk) [reply packages/raystack/calendar/date-picker.tsx:154]: We don't want user to overwrite the fields that are below {...calendarProps}

**Review by paanSinghCoder — APPROVED** [review]

**Pull request merged**